### PR TITLE
Signinrelatedupdates

### PR DESCRIPTION
**Pull request by noodlemoodle**

### CODE REVIEW

File: src/foam/nanos/u2/navigation/SignUp.js [hunk 1]
```diff
@@ -297,6 +295,22 @@ foam.CLASS({
...
           });
       }
+    },
+    {
+      name: 'footer',
+      section: 'footerSection',
+      label: 'Sign in',
+      buttonStyle: 'LINK',
+      code: function(X) {
+        window.history.replaceState(null, null, window.location.origin);
```
Comment (kgrgreer) [inline src/foam/nanos/u2/navigation/SignUp.js]: same

**Review by jlhughes — APPROVED** [review]

File: src/foam/u2/view/LoginView.js [hunk 2]
```diff
@@ -220,7 +227,17 @@ foam.CLASS({
...
+        try {
+          this.model = (foam.lookup(this.modelCls))?.create(this.param, this);
+          return;
+        } catch (err) {
+          console.warn('Error occurred when looking up modelCls', this.modelCls, err);
+        }
+      }      
+
       // Instantiating model based on mode_
       if ( this.mode_ === this.MODE1 ) {
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/LoginView.js:242]: Wouldnt this override the modelCls_?

File: src/foam/u2/view/LoginView.js [hunk 3]
```diff
@@ -242,7 +259,7 @@ foam.CLASS({
       // CREATE MODEL VIEW
       var right = this.E()
       // Header on-top of rendering model
-        .start().hide(this.imgPath$).addClass('topBar-logo-Back')
+        .start().show(this.imgPath$).addClass('topBar-logo-Back')
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/LoginView.js]: Why this change?

File: src/foam/u2/view/LoginView.js (hunk 2 above)
Comment (noodlemoodle) [reply src/foam/u2/view/LoginView.js:242]: this should be fine since there's no default value for mode_

File: src/foam/u2/view/LoginView.js (hunk 3 above)
Comment (noodlemoodle) [reply src/foam/u2/view/LoginView.js]: think i might have misunderstood what this was supposed to do, will change back, thanks

File: src/foam/nanos/u2/navigation/SignIn.js [hunk 4]
```diff
@@ -206,6 +197,30 @@ foam.CLASS({
...
         }
       }
+    },
+    {
+      name: 'footer',
+      label: 'Create an account',
+      section: 'footerSection',
+      buttonStyle: 'LINK',
+      code: function(X) {
+        window.history.replaceState(null, null, window.location.origin);
```
Comment (kgrgreer) [inline src/foam/nanos/u2/navigation/SignIn.js]: Generally, better to do X.window or import 'window' and do this.window. Accessing the global window prevents the code from working with multiple windows.
